File: src/ClimateProject.Domain/Entities/SystemSettings.cs
namespace ClimateProject.Domain.Entities;

public class SystemSettings
{
    public Guid Id { get; set; }
    public bool LoginEnabled { get; set; } = true;
    public bool MaintenanceMode { get; set; }
    public string? MaintenanceMessage { get; set; }
    public int MaxLoginAttempts { get; set; } = 5;
    public int SessionTimeoutMinutes { get; set; } = 60;
    public PasswordPolicy PasswordPolicy { get; set; } = new();
    public SystemEmailSettings EmailSettings { get; set; } = new();
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }
}

public class PasswordPolicy
{
    public int MinLength { get; set; } = 8;
    public bool RequireUppercase { get; set; } = true;
    public bool RequireLowercase { get; set; } = true;
    public bool RequireNumbers { get; set; } = true;
    public bool RequireSpecialChars { get; set; }
}

public class SystemEmailSettings
{
    public bool SmtpEnabled { get; set; }
    public string? FromEmail { get; set; }
    public string? SmtpHost { get; set; }
    public int? SmtpPort { get; set; }
}

File: web/src/features/org-structure/api/systemSettings.ts
import { authFetch } from '../../../api/authFetch'

export interface PasswordPolicy {
  minLength: number
  requireUppercase: boolean
  requireLowercase: boolean
  requireNumbers: boolean
  requireSpecialChars: boolean
}

export interface SystemEmailSettings {
  smtpEnabled: boolean
  fromEmail: string | null
  smtpHost: string | null
  smtpPort: number | null
}

export interface SystemSettingsData {
  loginEnabled: boolean
  maintenanceMode: boolean
  maintenanceMessage: string | null
  maxLoginAttempts: number
  sessionTimeoutMinutes: number
  passwordPolicy: PasswordPolicy
  emailSettings: SystemEmailSettings
  updatedAt: string
}

export interface UpdateSystemSettingsInput {
  loginEnabled?: boolean
  maintenanceMode?: boolean
  maintenanceMessage?: string
  maxLoginAttempts?: number
  sessionTimeoutMinutes?: number
  passwordPolicy?: PasswordPolicy
  emailSettings?: SystemEmailSettings
}

/**
 * `maintenanceMessage` is a paired column (#210) resolved server-side for `lang`. Without
 * it the server answered in its fallback language, and a Spanish operator read the English
 * half of a bilingual notice on the one screen that can turn sign-in off.
 */
export async function getSystemSettings(baseUrl: string, lang?: string): Promise<SystemSettingsData> {
  const query = lang ? `?lang=${encodeURIComponent(lang)}` : ''
  const response = await authFetch(`${baseUrl}/admin/system-settings${query}`)
  return response.json() as Promise<SystemSettingsData>
}

export async function updateSystemSettings(baseUrl: string, input: UpdateSystemSettingsInput): Promise<SystemSettingsData> {
  const response = await authFetch(`${baseUrl}/admin/system-settings`, {
    method: 'PUT',
    body: JSON.stringify(input),
  })
  return response.json() as Promise<SystemSettingsData>
}
